### DISCUSSION TITLE
Is _g_assert_property_notify  a redundant interface?

### DISCUSSION THREAD

**Discussion by MANGOPIE3** [storaged-project/udisks#1199] [General]
_g_assert_property_notify
What is the use of `_g_assert_property_notify` ? It's not called. Is it a redundant interface?
https://github.com/storaged-project/udisks/blob/2214c378862ba009be04cea5cede3c283297b785/src/tests/testutil.h#L4

**Comment by tbzatek** [comment]
No idea, you're looking at 13 years old code that nobody touched since. What's the problem?

**Reply by MANGOPIE3** [reply]
I found that abnormal_termination test  called it if compiled tests with clang and with -O2 Optimized option! It was very strange.
But I've figured it out. 
Clang -O2 will delete the derefering the null pointer operation (gcc not) and incorrectly call the _g_assert_property_notify_run interface.
![image](https://github.com/storaged-project/udisks/assets/17828807/b46dc2b2-51af-4edc-a3d2-a9c5a0b02a92)

**Reply by tbzatek** [reply]
Anyway, feel free to submit a pull request, we have no immediate plans to change this code.
